src/client/metadataTransfer.ts

```diff
@@ -269,3 +269,16 @@ export abstract class MetadataTransfer<Status extends MetadataRequestStatus, Res
   protected abstract pre(): Promise<AsyncResult>;
   protected abstract post(result: Status): Promise<Result>;
 }
+
+/* prevent requests on apiVersions higher than the org supports */
+const getConnectionNoHigherThanOrgAllows = async (conn: Connection, requestedVersion: string): Promise<Connection> => {
```

Comment:
Do we have test coverage for this method?

Reply:
it's being executed on every deploy, so about 100x.
At least some of those specify an apiVersion...not sure if any don't and would check the if conditional.